openedx/core/djangoapps/content/block_structure/store.py

```diff
@@ -207,7 +207,15 @@ def _deserialize(self, serialized_data, root_block_usage_key):
         """
         Deserializes the given data and returns the parsed block_structure.
         """
-        block_relations, transformer_data, block_data_map = zunpickle(serialized_data)
+
+        try:
+            block_relations, transformer_data, block_data_map = zunpickle(serialized_data)
+        except Exception:
+            # Somehow failed to de-serialized the data, assume it's corrupt.
+            bs_model = self._get_model(root_block_usage_key)
+            logger.warning("BlockStructure: Failed to load data from cache for %s", bs_model)
+            raise BlockStructureNotFound(bs_model.data_usage_key)
```

Comment:
If we're doing wholesale exception swallowing like this, can we log the exception explicitly? I'm just worried we'll get some weird unicode error or something mixed in there that we'll have no visibility into.

Reply:
Done.